store.rb

```diff
@@ -118,4 +148,39 @@ def get_node(node)
       raise e
     end
   end
+
+  # immediately update cache if node joins/leaves
+  def watch()
+    return if @zk.nil?
+    @root_watcher ||= @zk.register("/", :only => :child) do |event|
+      @log.info "Children added/deleted"
+      reload_instances
+    end
+  end
+
+  def reload_instances()
+    start_seq = 0
+    @mutex.synchronize do
+      @sequence += 1
+      start_seq = @sequence
+    end
+    instances = load_instances_from_zk()
+    @mutex.synchronize do
+      if start_seq > @current_seq then
+        @current_seq = start_seq
+        @cached_results = instances
+      end
+    end
+  end
+
+  def start_fetch_thread()
+    @should_run = true
```

Comment:
what's the point of changing `@should_run` in so many places? also, can you name it something more descriptive, like `fetch_thread_should_run` ?